### PR TITLE
Fix searchHighlight default formatter

### LINKED ISSUE #5396: Search highlight not working when type of the data field is number
**Bootstraptable version(s) affected**:
1.18.0

**Description**  
search highlight

**Example**  
https://examples.bootstrap-table.com/#options/search-highlight.html
Uncaught TypeError: c.replace is not a function
    at bootstrap-table.min.js:10
    at Array.forEach (<anonymous>)
    at e.value (bootstrap-table.min.js:10)
    at e.value (bootstrap-table.min.js:10)
    at e.value (bootstrap-table.min.js:10)
    at e.value (bootstrap-table.min.js:10)
    at bootstrap-table.min.js:10

**Possible  (optional)**  
Could this error be caused by line 1563 in [bootstrap-table.js ](https://github.com/wenzhixin/bootstrap-table/blob/develop/src/bootstrap-table.js) ?
Possible fixed: change 'value.replace(...)' to 'value.toString().replace(...)' ?

**Additional context**  
[data1.json](https://examples.bootstrap-table.com/json/data1.json) contains non-string fields such as "id", "amount"

### PR DESCRIPTION
**Pull request by Moutard3**
**Bug fix?**
yes

**New Feature?**
no

**Resolve an issue?**
Fix #5396 

**Example(s)?**
Before: https://live.bootstrap-table.com/example/options/search-highlight.html
After: https://live.bootstrap-table.com/code/Moutard3/5157 (moutard3:fix/5396)

### CODE REVIEW

**Review by djhvscf — CHANGES_REQUESTED** [review]
Please review Travis errors

File: src/bootstrap-table.js [hunk 1]
```diff
@@ -1602,9 +1605,6 @@ class BootstrapTable {
 
         item[this.header.stateField] = value === true || (!!value_ || (value && value.checked))
       } else {
-        value = typeof value === 'undefined' || value === null ?
```
Comment (UtechtDustin) [inline src/bootstrap-table.js:1605]: why do you removed that ?

Comment (Moutard3) [reply src/bootstrap-table.js:1605]: removed as it is now on line 1562 as we need value to be !== null for line 1566

**Comment by djhvscf** [comment]
This is working as expected. @UtechtDustin please review it and give us your approval

**Review by djhvscf — APPROVED** [review]

**Comment by UtechtDustin** [comment]
@djhvscf sorry that i dont answered im currently busy (check mattermost)

**Pull request merged**